### ISSUE TITLE
Router: Audit ASSERTs

### ISSUE THREAD

**Issue by asraa** [envoyproxy/envoy#14961]
This issue describes a scan of ASSERTs in the router filter and recommendations to change some: https://gist.github.com/asraa/859a41a8f5f252d35da58c9ef7502230

Those marked with a verdict have been audited or unknown if they need a pair of eyes.

- [ ] Change 1 refactorable call https://github.com/envoyproxy/envoy/blob/488973532ac72f56473bd2b6f770f0a011bf14a1/source/common/router/router.cc#L276 add check for scheme headers during filter processing https://github.com/envoyproxy/envoy/blob/488973532ac72f56473bd2b6f770f0a011bf14a1/source/common/router/router.cc#L566
- [ ] Audit 
  * https://github.com/envoyproxy/envoy/blob/488973532ac72f56473bd2b6f770f0a011bf14a1/source/common/router/router.cc#L653

**Comment by daixiang0** [comment]
@asraa is it still available?

**Comment by asraa** [comment]
Yes I think so -- feel free! to take on one of those two points.
